### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by Chulstro**
Encryption functionality completed

### CODE REVIEW

File: encrypt.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+file = gets.chomp
+text = File.open(file, "r")
+message = text.read
+enigma = Enigma.new
+puts "Name a destination for your encryption"
+new_file = gets.chomp
+encryption = File.open(new_file, 'w')
+encryption.write(enigma.encrypt(message)[:encryption])
+
+puts "Created #{new_file} with the key of #{enigma.key} and date of #{enigma.date}"
```
Comment (Chulstro) [inline encrypt.rb:13]: Runner file that allows a command line interaction with the encryption code

File: encryption.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+qftvx,
+
+vzhxjnmjrohswjoyinwxbpfk.afycassummninfjobarns.jysmzjsmjbphnrf.
+
+dwamjivuz,
+afmxibjydf
```
Comment (Chulstro) [inline encryption.txt:6]: Encryption file that holds the encryption of the message txt file

File: lib/character.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+class Character
+  attr_reader :id
+
+  def initialize(id)
+    @id = id
+  end
+end
```
Comment (Chulstro) [inline lib/character.rb:8]: Created a class so that an index could be created without worrying about repeating characters

File: lib/enigma.rb [hunk 4]
```diff
@@ -11,18 +12,25 @@ def self.characters
...
-    finder = (@date.to_i ** 2).to_s
-    @offset = format('%04d',finder[-4..-1])
+    @message = Message.new(message, @shift)
+    accumulator[:encryption] = @message.change_message
+    accumulator[:key] = @key
+    accumulator[:date] = @date
+    accumulator
   end
+
 end
```
Comment (Chulstro) [inline lib/enigma.rb:36]: Create an accumulator to create a final hash for encrypt method.  Integrates both message and shift classes to make shift hash and encrypt given strings.

File: lib/message.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        elsif (@message_array.index(character) + 2) % 4 == 0
+          find_new_char.rotate(@shift[:C])[0]
+        else
+          find_new_char.rotate(@shift[:D])[0]
+        end
+      end
+    end.join
+  end
+
+end
```
Comment (Chulstro) [inline lib/message.rb:38]: Turns every character in a message into an array of characters and then turns each character into an object that can be indexed. This then iterates through each index to shift it the appropriate amount. Then combines them together to make a new string.

File: message.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+Hello,
+
+My name in Inigo Montoya. You killed my father. Prepare to die.
+
+Cordially,
+Seen above
```
Comment (Chulstro) [inline message.txt:6]: Example text file

File: test/character_test.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+class CharacterTest < Minitest::Test
+
+  def setup
+    @a = Character.new("a")
+  end
+
+  def test_it_can_hold_character
+    assert_equal "a", @a.id
+  end
+end
```
Comment (Chulstro) [inline test/character_test.rb:13]: Character class needed to make the index work

File: test/enigma_test.rb [hunk 8]
```diff
@@ -4,11 +4,17 @@
...
-    @message.encrypt("Fight me")
+    expected = {
+      encryption: "keder ohulw",
+      key: "02715",
+      date: "040895"
+    }
     assert_equal "a", Enigma.characters[0]
+    assert_equal expected, @enigma.encrypt("hello world", "02715", "040895")
   end
 end
```
Comment (Chulstro) [inline test/enigma_test.rb:20]: Updated to produce a hash for encrypt method

File: test/message_test.rb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    assert_equal ["h", "e", "l", "l", "o", " ", "w", "o", "r", "l", "d"], @message.message_array
+    assert_equal expected, @message.shift
+  end
+
+  def test_it_finds_index
+    assert_equal "k", @message.move_alphabet("k")[0]
+    assert_equal "keder ohulw", @message.change_message
+    assert_equal "keder ohulw!", @message_2.change_message
+  end
+end
```
Comment (Chulstro) [inline test/message_test.rb:30]: Test for message class and gets a couple edge cases

**Comment by Chulstro** [comment]
Needs more comprehensive testing but functionality first
